### Задание 1.
Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти ширину доверительного интервала для оценки математического ожидания `mu` с надёжностью 0.95, если выборочное среднее равно 80, а объём выборки равен 256.

In [1]:
from scipy import stats
import numpy as np

In [2]:
std = 16
alpha = 0.05
Mx = 80
n = 256
Zhalfalpha = stats.norm.ppf(alpha / 2)

In [3]:
a = Mx+Zhalfalpha*(std/(256**0.5))
a

78.04003601545995

In [4]:
b = Mx-Zhalfalpha*(std/(256**0.5))
b

81.95996398454005

In [5]:
interval = b-a
interval

3.9199279690801063

##### Ответ: 3.91992797 

### Задание 2

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%? Найдите P-значение для данного теста и укажите его в качестве ответа.

In [6]:
alpha = 0.01
x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
Mx = 200
std = np.std(x, ddof=1)
n = x.shape[0]
t = (x.mean() - Mx) / (std/n**0.5)



In [7]:
p_left = stats.t.cdf(t, df=n - 1)
p_right = 1 - stats.t.cdf(t, df=n - 1)

pvalue = 2 * min(p_left, p_right)
pvalue

0.31456660352472854

##### Ответ: 0.31456660

### Задание 3

Дана выборка диаметров подшипников из примера 1 с занятия 5:
```
samples = [0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
           1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
           0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ]
```
Предполагая, что диаметры подшипников распределены нормально, проверьте гипотезу о том, что дисперсия случайной величины равна `0.0625` при уровне значимости `alpha = 0.05`. Что для этого нужно знать:
1. Альтернативная гипотеза двухсторонняя.
2. Статистика для теста: `H = (n - 1) * sample_variance / variance`, где `n` - число элементов в выборке, `sample_variance` - несмещённая оценка дисперсии, `variance` - утверждаемая нулевой гипотезой дисперсия.
3. Эта статистика в предположении верности нулевой гипотезы имеет распределение хи-квадрат с параметром `df = n - 1`. Её квантили можно найти с помощью функции `scipy.stats.chi2.ppf`.

В качестве ответа запишите 1, если нулевая гипотеза верна, и 0 иначе.

In [8]:
samples = np.array([0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
           1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
           0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ])
alpha = 0.05
variance = 0.0625
sample_variance = np.var(samples, ddof=1)
n = samples.shape[0]

In [9]:
H = (n - 1) * sample_variance / variance
H

28.720541753333343

In [10]:
t1 = stats.chi2.ppf(alpha/2, df = n - 1)
t2 = stats.chi2.ppf(1-alpha/2, df = n - 1)
t1,t2

(11.688551922452438, 38.0756272503558)

- Критическая область = (-$∞$, 11.688551922452438) U (38.0756272503558, $∞$)
- Статистика для теста H = 28.720541753333343

Статистика для теста НЕ лежит в критической области, следовательно нулевая гипотеза верна.

#### Ответ: 1